packages/stylist-core/src/ast/rule_content.rs

```diff
@@ -10,7 +11,7 @@ pub enum RuleContent {
     /// A block
     Block(Block),
     /// A nested rule
-    Rule(Box<Rule>),
+    Rule(Bow<'static, Rule>),
```

Comment:
The moment a nested item isn't `const`, it forces all parents to use a `vec![]` instead of the `const` array, aswell, i.e. they will allocate. Now, given nested rules aren't that common, but they do appear, especially with `@supports`.

Reply:
Whilst I suspect the following should work, the compiler actually panics when I change the type to `Cow<'static, Box<T>>`:

```rust
{
    use ::stylist::vendor::once_cell::sync::Lazy;
    static RULE: Lazy<Box<Rule>> = Lazy::new(|| Box::new(#rule_tokens));
    Cow::Borrowed(RULE)
}
```

This is the first time I actually encountered an ICE. :)

```
error: internal compiler error: compiler/rustc_traits/src/normalize_erasing_regions.rs:54:32: could not fully normalize `<std::boxed::Box<ast::rule::Rule> as std::borrow::ToOwned>::Owned`

thread 'rustc' panicked at 'Box<dyn Any>', compiler/rustc_errors/src/lib.rs:1034:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

note: the compiler unexpectedly panicked. this is a bug.
```

I also tested `Arc` by replacing all `const` with `Lazy` and achieved a similar performance. But the generated bundle is  slightly bigger. I guess the best course of action for now is to keep `Bow`.

However, `stylist::vendor` is reserved for vendored third party libraries (which should be removed once `std::lazy` is stabilised). Maybe move `Bow` to `stylist::bow` or `stylist::utils`?

Reply:
I think moving it under `stylist::macros` makes sense, which is afaik how `std::fmt` does it with things like `Arguments`.

Reply:
I am not very satisfied with the reexport at the root of the documentation.

And after referencing the documentation of `std`, `tokio`, `yew`, `syn` and some other libraries, I think the reasonable action is to remove `macros` module and move macros back to the root of the library.

Which I will do when after the PR list is cleared, so please move `Bow` out of `macros` modules.

Reply:
> I think the reasonable action is to remove macros module 

I think originally I introduced it to have a vehicle to carry the documentation of the syntax so that it doesn't have to be repeated over and over in every single macro. That's also why `tokio::stream` exists. I think it serves that purpose better than it serves as a host for the macro re-exports. Having all the macro exports at top-level would make sense to me though. Lastly, I think the `macros` module serves well as a place to host `vendor` and other things that are used in macro expansion, since it should be `#[cfg(feature = "macros")]`, thus then also turning off the vendored re-exports?

```
stylist
├── sheet!
├── css!
├── global_style!
├── style!
└── macros
    ├── #doc
    ├── #[doc(hidden)] vendor
    │   └── pub use ...
    └── #[doc(hidden)] Bow
```

Reply:
Maybe move Bow to ast?

Bow is currently used by AST regardless of whether feature macros is enabled.

Unless we have feature flags on AST to control which Boxing is used (which I think it's not worth it).

In addition, Bow is also actually visible in the AST(RuleContent::Rule), so I think Bow itself also needs to be visible in the documentation.

As for vendor, I think both stylist::vendor and stylist::macros::vendor are fine.